### PR TITLE
chore: adds beta opt in analytics event

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ksopt-in-analytics-event-hashicorp.vercel.app/waypoint/tutorials) 🔎
- [Asana task](https://app.asana.com/0/1199634971449915/1202207336073938) 🎟️

## 🗒️ What

This PR adds a new `useOptInAnalyticsTracking` hook, which can be called per view to handle the firing of the opt in event. The hook checks for the presence of the platform opt-in cookie and fires the event if it hasn't already been done. Once the event is fired, a new cookie is created to note that this opt-in event was tracked.

### CODE REVIEW

File: src/layouts/base-new/index.tsx [hunk 1]
```diff
@@ -40,7 +44,12 @@ const BaseNewLayout: React.FC<BaseNewLayoutProps> = ({
           )}
         </div>
       </CoreDevDotLayout>
-      <ConsentManager />
+      <ConsentManager
+        onAcceptAll={() => {
+          handleOptInAnalytics(query['optInFrom'] as OptInPlatformOption)
+          console.log('accepted!!')
```
Comment (kendallstrautman) [inline src/layouts/base-new/index.tsx:50]: TODO - it doesn't seem like this function is firing as expected. I need to debug within the consent manager 🤔

Comment (ashleemboyer) [reply src/layouts/base-new/index.tsx:50]: I got curious and checked this out myself, it looks like you have to pass the props through the object that's passed to `createConsentManager`! I tried it out and that worked for me. Although, I was expecting your approach here to work as well, and it feels weird that it doesn't work.

Comment (kendallstrautman) [reply src/layouts/base-new/index.tsx:50]: Oh interesting! Okay I'll try that. Ty 😄

Comment (ashleemboyer) [reply src/layouts/base-new/index.tsx:50]: Sure thing! 😊

File: src/views/collection-view/index.tsx [hunk 2]
```diff
@@ -20,6 +21,7 @@ function CollectionView({
   layoutProps,
   product,
 }: CollectionPageProps): React.ReactElement {
+  useOptInAnalyticsTracking('learn')
```
Comment (kendallstrautman) [inline src/views/collection-view/index.tsx:24]: The analytics hook is called only on the views that handle opt-out UI. We could eventually lift this up, but this feels appropriate for now.

**Review by BrandonRomano — APPROVED** [review]
This looks good to me; I don't have any concerns about perf from that hook as it should be minimal; and it's temporary so this is good from my perspective!

**Pull request merged**